conans/client/rest/conan_requester.py

```diff
@@ -61,6 +66,22 @@ def _get_auth(credentials):
             raise ConanException(f"Error loading 'source_credentials.json' {creds_path}: {repr(e)}")
 
     def add_auth(self, url, kwargs):
+        # First, try to use "auth_source_plugin"
+        if self.auth_source_plugin:
+            try:
+                c = self.auth_source_plugin(url)
+            except Exception as e:
+                msg = f"Error while processing 'auth_source_remote.py' plugin"
+                msg = scoped_traceback(msg, e, scope="/extensions/plugins")
+                raise ConanException(msg)
+            if c:
```

Comment:
If we expect a dict-like object, why do we not return this one for all the statements? So we can remove this `if` and directly pass it to the next statement `if c.get("token"):`.

Reply:
The plugin can give `None` in order to give a fallthrough.

Reply:
Where are we checking that `None` value?

Reply:
We are checking that on the `test_auth_source_plugin_fallback`